mlflow/dspy/callback.py

```diff
@@ -370,3 +382,24 @@ def _unpack_kwargs(self, inputs: dict[str, Any]) -> dict[str, Any]:
         kwargs = inputs.get("kwargs", {})
         inputs_wo_kwargs = {k: v for k, v in inputs.items() if k != "kwargs"}
         return {**inputs_wo_kwargs, **kwargs}
+
+    def _generate_result_table(
+        self, outputs: list[tuple[dspy.Example, dspy.Example, Any]]
+    ) -> dict[str, list[Any]]:
+        result = defaultdict(list)
+        for i, (example, prediction, score) in enumerate(outputs):
+            for k, v in example.items():
+                result[f"example_{k}"].append(v)
+
+            if not hasattr(prediction, "items"):
```

Comment:
we can probably just do `if instance(prediction, dspy.Prediction)` for clarity

Reply:
that's a weird style... I think it's safe to assume it's either a dict or `dspy.Prediction`, and error out when that's other types.